Lib/asyncio/proactor_events.py

```diff
@@ -159,6 +168,11 @@ def pause_reading(self):
         if self._closing or self._paused:
             return
         self._paused = True
+
+        if self._read_fut is not None and not self._read_fut.done():
```

Comment:
Proactor tricks again?
Not sure if cancelling is what we should do here.
`self._read_fut` should be set on next `resume_reading` call maybe? No sure though.

Reply:
No, I think this is just a bug.  The problem is that when you call `pause_reading`, the current code doesn't actually pause it. It just sets a `_paused` flag but leaves the socket in the reading mode. Do you agree, or am I missing something here?

Reply:
Looks like you are right